packages/accounts/src/kernel-zerodev/middleware/gas-estimator.ts

```diff
@@ -14,28 +14,38 @@ import { calcPreVerificationGas } from "../utils/calc-pre-verification-gas.js";
 export const withZeroDevGasEstimator = (
   provider: ZeroDevProvider
 ): ZeroDevProvider => {
-  provider.withFeeDataGetter(async () => {
+  provider.withFeeDataGetter(async (struct) => {
+    let overrides = await resolveProperties({
+      maxFeePerGas: struct.maxFeePerGas ?? 0n,
+      maxPriorityFeePerGas: struct.maxPriorityFeePerGas ?? 0n,
+    });
+
+    let maxFeePerGas, maxPriorityFeePerGas;
+
     try {
-      let { maxFeePerGas, maxPriorityFeePerGas } = await eip1559GasPrice(
+      ({ maxFeePerGas, maxPriorityFeePerGas } = await eip1559GasPrice(
         provider
-      );
-      return {
-        maxFeePerGas,
-        maxPriorityFeePerGas,
-      };
+      ));
     } catch (error: any) {
       console.warn(
         "getGas: eth_maxPriorityFeePerGas failed, falling back to legacy gas price."
       );
     }
 
-    const feeData = await getFeeData(provider);
-    const maxFeePerGas = feeData?.maxFeePerGas
-      ? BigInt(feeData?.maxFeePerGas)
-      : 0n;
-    const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas
-      ? BigInt(feeData.maxPriorityFeePerGas)
-      : 0n;
+    if (maxFeePerGas === undefined || maxPriorityFeePerGas === undefined) {
+      const feeData = await getFeeData(provider);
+      maxFeePerGas = feeData?.maxFeePerGas ? BigInt(feeData?.maxFeePerGas) : 0n;
+      maxPriorityFeePerGas = feeData.maxPriorityFeePerGas
+        ? BigInt(feeData.maxPriorityFeePerGas)
+        : 0n;
+    }
+
+    if (
+      maxFeePerGas < BigInt(overrides.maxFeePerGas) ||
+      maxPriorityFeePerGas < BigInt(overrides.maxPriorityFeePerGas)
+    ) {
+      return overrides;
```

Comment:
If `overrides.maxFeePerGas` > maxFeePerGas, but maxPriorityFeePerGas > `overrides.maxPriotyFeePerGas` should imo only use the override value for `maxFeePerGas` but not `maxPriorityFeePerGas`. Also, what if the overrides only specify one value?

Reply:
In this case, overrides are actually the increased fees from the retry logic. So it will always have both values.
There's one [TODO] to add the gas overrides when doing individual tx. For that though, I might have to do a PR on `@alchemy/core` or else I was going to include it in this PR. I will refine this logic when that PR is done.

Also, both variables by related to each other by below logic mostly, so it's safe to assume is one is larger than other one is as well:
`const maxFeePerGas = block.baseFeePerGas * 2n + maxPriorityFeePerGas`